spec/acts_as_taggable_on/tag_spec.rb

```diff
@@ -171,7 +171,7 @@
     context 'retry 3 times on not unique exception' do
       it 'performs 3 tries before raising the exception' do
         allow(ActsAsTaggableOn::Tag).to receive(:named_any).and_raise(ActiveRecord::RecordNotUnique.new('error')) # trigger error inside block
-        expect(ActiveRecord::Base.connection).to receive(:execute).with('ROLLBACK').exactly(3).times
+        expect(ActiveRecord::Base.connection).to receive(:execute).with('ROLLBACK').exactly(4).times # one extra rollback from database cleaner
```

Comment:
Let change the Datacleaner strategy to :deletion and correct the test 
https://github.com/mbleigh/acts-as-taggable-on/blob/master/spec/support/database_cleaner.rb#L5